plugins/inputs/snmp/snmp.go

```diff
@@ -535,8 +538,8 @@ func (t Table) Build(gs snmpConnection, walk bool) (*RTable, error) {
 		Time: time.Now(), //TODO record time at start
 		Rows: make([]RTableRow, 0, len(rows)),
 	}
-	for _, r := range rows {
-		if len(r.Tags) < tagCount {
+	for i, r := range rows {
+		if len(r.Tags)+emptyTags[i] < tagCount {
 			// don't add rows which are missing tags, as without tags you can't filter
 			continue
 		}
```

Comment:
I'm really confused. The only difference between `rtr4` and `rtr3` is the missing `myfield2` which isn't even a tag. If you are proposing that we drop based on a missing field, instead of dropping based on a missing tag, then both `rtr3` and `rtr4` would be dropped as they're both missing `myfield4`